pages/[username].tsx

```diff
@@ -0,0 +1,198 @@
+import Image from "next/image";
+import LayoutContainer from "@components/LayoutContainer";
+import RedirectUtils from "utils/redirect";
+import Toggle from 'react-toggle'
+import UserAgentUtils from "utils/userAgent";
+import getConfig from "next/config";
+import parser from 'ua-parser-js'
+import useToastHook from "@hooks/toast/useToastHook"
+import React, { useEffect, useRef, useState } from "react";
+import { BaseContainer } from "@components/Page/BaseContainer";
+import { GetServerSidePropsContext } from "next";
+import { Header } from "@components/Header";
+import { Helmet } from "react-helmet";
+import { LoaderWrapper } from "@components/LoaderWrapper";
+import { MessageEnum } from "@shared/enum";
+import { toast } from "react-toastify";
+import { useGetProfile } from "@services/profile/profileHooks";
+import { useRouter } from "next/router";
+
+interface PageProps {
+    username?: string
+    isDynamicLink?: boolean
+}
+
+const { publicRuntimeConfig } = getConfig()
+
+export default function Profile(props: PageProps) {
+    const {isDynamicLink, username} = props
+    const { data, isLoading } = useGetProfile(username);
+    const [message, setMessage] = useState('')
+    const router = useRouter();
+    useEffect(() => {
+        if (data) {
+            localStorage.setItem(MessageEnum.targetUser, data.user_id);
+        }
+    }, [data])
+
+    useEffect(() => {
+        if (isDynamicLink) router.push(`/${username}`, undefined, { shallow: true })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isDynamicLink])
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You are using `router.push` inside a `useEffect` hook without including `router` and `username` in the dependency array. This could lead to unexpected behavior if these values change.

```diff
    useEffect(() => {
        if (isDynamicLink) router.push(`/${username}`, undefined, { shallow: true })
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [isDynamicLink, router, username])
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->